pkg/cloudprovider/providers/gce/gce_instances.go

```diff
@@ -273,35 +274,42 @@ func (gce *GCECloud) AddSSHKeyToAllInstances(user string, keyData []byte) error
 	})
 }
 
-// GetAllZones returns all the zones in which nodes are running
-func (gce *GCECloud) GetAllZones() (sets.String, error) {
+// GetAllCurrentZones returns all the zones in which k8s nodes are currently running
+func (gce *GCECloud) GetAllCurrentZones() (sets.String, error) {
+	if gce.nodeInformerSynced == nil {
+		glog.Warningf("GCECloud object does not have informers set, should only happen in E2E tests.")
+		return gce.GetAllZonesE2E()
```

Comment:
GetAllZonesE2E should not exist in non-test code

Reply:
why doesn't looking up from the node labels work in e2e? And why is nodeInformerSynced nil in e2e?

Reply:
In the E2E tests it seems that there is a "local" version of the cloud provider that doesn't have any informers set (so checking nodeInformerSynced nil is a good proxy for checking existence of informers). We need "backup" GetAllZones functionality for the tests that use GetAllZones functionality with this local cloud provider controller. Maybe GetAllZonesE2E isn't the best name for it?